hack/builder/Dockerfile

```diff
@@ -130,6 +142,20 @@ COPY entrypoint.sh /entrypoint.sh
 
 COPY create_bazel_cache_rcs.sh /create_bazel_cache_rcs.sh
 
+#Bazel support for s390x is not available in upstream
+#Copy the pre-built bazel-s390x binary to builder image
+
+RUN  mkdir -p /usr/s390x-bazel
+
+COPY ./bazel /usr/s390x-bazel/
```

Comment:
Doesn't this command need to be run only on s390x?I suppose we don't want to copy the binary also on the arm and x86 image. However, I'm not sure how you can conditionally run the copy commad in a Dockerfile

Reply:
Additionally, where is this binary fetched locally?

Reply:
It is a bit convoluted but I like the suggestion from here for a real conditional copy: https://stackoverflow.com/questions/31528384/conditional-copy-add-in-dockerfile

You could create 2 intermediate building steps with the command you used below for fetching bazel and then copy the binary from the intermediate step. I hope it is clear

Reply:
@vamsikrishna-siddu 

Please put appropriate condition for s390x arch.

Reply:
I think either making it a multistage build, or alternatively putting the s390x build into a separate Dockerfile are both options to consider. Especially since the long term strategy would be to have bazel on s390x, making this a hopefully temporary solution.
When splitting it into a separate Dockerfile and/or multistage build, it might also be good to include the building bazel part as a stage. This could be easily done using https://github.com/linux-on-ibm-z/scripts/blob/master/Bazel/5.3.2/build_bazel.sh (needs an ubuntu stage). That way the actual logic for building bazel is not maintained inside the kubevirt project, we would just be fetching and adapting an existing build script.

Reply:
Please, try to minimize the code changes. So, I would avoid a second Dockerfile, I would be in favor of multi stage building. Something like:
```dockerfile
ARG BUILD_ARCH

FROM <build-bazel>:s390x as bazel-s390x
# Fetch here the bazel versio 


FROM quay.io/centos/centos:stream9 as bazel-default
curl -L -o /usr/bin/bazel https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-linux-${BAZEL_ARCH} && chmod u+x /usr/bin/bazel;

# In the last stage
COPY --from=bazel-${BUILD_ARCH} /usr/bin/bazel /usr/bin/bazel
```

Then, we can poss `--build-arg=s390x` or `--build-arg=default` and select the stage based on the arch

Reply:
I think it meight be better to put the condition logic into a single bazel stage, and just do
```
COPY --from=bazel /bazel /usr/bin/bazel
```
instead. That way we can just use the normal arch arg and don't need a new one.

Reply:
This work too if you already have the bazel binary ready somewhere to fetch. Is this the case?

Reply:
Unfortunately not, but i was thinking more about 2 separate `RUN` steps, one if arch is s390x, one if not.

Reply:
> Additionally, where is this binary fetched locally?
@alicefr  I have included the bazel at this place hack/builder/bazel 
https://github.com/kubevirt/kubevirt/pull/11097/files#diff-a3062a93f41b73857652d6d8edacd4b84e21c40ddca7b74189bc7845ab0e630a

Reply:
> It is a bit convoluted but I like the suggestion from here for a real conditional copy (second answer): https://stackoverflow.com/questions/31528384/conditional-copy-add-in-dockerfile
> 
> You could create 2 intermediate building steps with the command you used below for fetching bazel and then copy the binary from the intermediate step based on the ARCH. I hope it is clear

Hi @alicefr  I have followed your suggestion and implemented a conditional-copy-add that will copy the bazel binary if the architecture is s390x else it will fetch the binary from the github releases.
you can see my latest commit.
https://github.com/kubevirt/kubevirt/pull/11097/commits/8817df01bca5819b69a23238f634ea09a3937632